### ISSUE TITLE
1.63 Beta fails to mount on macOS with on-the-fly crypt remote 

### ISSUE THREAD

**Issue by Jwink3101** [rclone/rclone#7063]
#### The associated forum post URL from `https://forum.rclone.org`

https://forum.rclone.org/t/1-63-beta-fails-to-mount-on-macos-with-on-the-fly-crypt-remote/39090/1

#### What is the problem you are having with rclone?

Using the latest 1.63 beta, rclone crashes (and crashes FUSE which locks the system) when trying to mount an on-the-fly crypt remote.

#### What is your rclone version (output from `rclone version`)

```test
rclone v1.63.0-beta.7072.4f8dab8bc
- os/version: darwin 13.4 (64 bit)
- os/kernel: 22.5.0 (arm64)
- os/type: darwin
- os/arch: arm64 (ARMv8 compatible)
- go/version: go1.20.5
- go/linking: dynamic
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

macOS 13.4 (ARM)

#### Which cloud storage system are you using? (e.g. Google Drive)

crypt and local

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

I worked my way up to try to find what break it. Note that rclone1.63-beta is the latest beta. These passwords are just for testing

The following WORKS: (See config below)

    rclone1.63-beta mount -vv mycrypt: ~/Desktop/mnt/

FAILS:

    export RCLONE_CRYPT_REMOTE='/Users/**USER**/rclone_scratch/cryptfiles/'
    export RCLONE_CRYPT_PASSWORD=rCd8GEoJZ3WQTk6aNRYop2gpwQ
    export RCLONE_CRYPT_PASSWORD2=ue8qsRrJrBLWKr5q3CWn6ake9w
    
    rclone1.63-beta mount -vv :crypt: ~/Desktop/mnt/

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2023/06/16 11:11:48 DEBUG : Setting --config "/Users/**USER**/rclone_scratch/config.cfg" from environment variable RCLONE_CONFIG="/Users/**USER**/rclone_scratch/config.cfg"
2023/06/16 11:11:48 DEBUG : Setting default for crypt-remote="/Users/**USER**/rclone_scratch/cryptfiles/" from environment variable RCLONE_CRYPT_REMOTE
2023/06/16 11:11:48 DEBUG : Setting default for crypt-password="rCd8GEoJZ3WQTk6aNRYop2gpwQ" from environment variable RCLONE_CRYPT_PASSWORD
2023/06/16 11:11:48 DEBUG : Setting default for crypt-password2="ue8qsRrJrBLWKr5q3CWn6ake9w" from environment variable RCLONE_CRYPT_PASSWORD2
2023/06/16 11:11:48 DEBUG : rclone: Version "v1.63.0-beta.7072.4f8dab8bc" starting with parameters ["rclone1.63-beta" "mount" "-vv" ":crypt:" "/Users/**USER**/Desktop/mnt/"]
2023/06/16 11:11:48 DEBUG : Creating backend with remote ":crypt:"
2023/06/16 11:11:48 DEBUG : Using config file from "/Users/**USER**/rclone_scratch/config.cfg"
2023/06/16 11:11:48 DEBUG : Setting crypt_remote="/Users/**USER**/rclone_scratch/cryptfiles/" from environment variable RCLONE_CRYPT_REMOTE
2023/06/16 11:11:48 DEBUG : Setting crypt_password="rCd8GEoJZ3WQTk6aNRYop2gpwQ" from environment variable RCLONE_CRYPT_PASSWORD
2023/06/16 11:11:48 DEBUG : Setting crypt_password2="ue8qsRrJrBLWKr5q3CWn6ake9w" from environment variable RCLONE_CRYPT_PASSWORD2
2023/06/16 11:11:48 DEBUG : :crypt: detected overridden config - adding "{vOUl7}" suffix to name
2023/06/16 11:11:48 DEBUG : Setting crypt_remote="/Users/**USER**/rclone_scratch/cryptfiles/" from environment variable RCLONE_CRYPT_REMOTE
2023/06/16 11:11:48 DEBUG : Setting crypt_password="rCd8GEoJZ3WQTk6aNRYop2gpwQ" from environment variable RCLONE_CRYPT_PASSWORD
2023/06/16 11:11:48 DEBUG : Setting crypt_password2="ue8qsRrJrBLWKr5q3CWn6ake9w" from environment variable RCLONE_CRYPT_PASSWORD2
2023/06/16 11:11:48 DEBUG : Creating backend with remote "/Users/**USER**/rclone_scratch/cryptfiles/"
2023/06/16 11:11:48 DEBUG : fs cache: renaming cache item "/Users/**USER**/rclone_scratch/cryptfiles/" to be canonical "/Users/**USER**/rclone_scratch/cryptfiles"
2023/06/16 11:11:48 DEBUG : fs cache: renaming cache item ":crypt:" to be canonical ":crypt,password='rCd8GEoJZ3WQTk6aNRYop2gpwQ',password2='ue8qsRrJrBLWKr5q3CWn6ake9w',remote='/Users/**USER**/rclone_scratch/cryptfiles/':"
2023/06/16 11:11:48 INFO  : Encrypted drive ':crypt{vOUl7}:': poll-interval is not supported by this remote
2023/06/16 11:11:48 DEBUG : Mounting on "/Users/**USER**/Desktop/mnt/" ("crypt,password='rCd8GEoJZ3WQTk6aNRYop2gpwQ',password2='ue8qsRrJrBLWKr5q3CWn6ake9w',remote=' Users **USER** rclone_scratch cryptfiles '")
2023/06/16 11:11:48 DEBUG : Adding "-o modules=iconv,from_code=UTF-8,to_code=UTF-8-MAC" for macOS
2023/06/16 11:11:48 DEBUG : Encrypted drive ':crypt{vOUl7}:': Mounting with options: ["-o" "attr_timeout=1" "-o" "fsname=:crypt,password='rCd8GEoJZ3WQTk6aNRYop2gpwQ',password2='ue8qsRrJrBLWKr5q3CWn6ake9w',remote='/Users/**USER**/rclone_scratch/cryptfiles/':" "-o" "subtype=rclone" "-o" "max_readahead=131072" "-o" "atomic_o_trunc" "-o" "daemon_timeout=600" "-o" "volname=crypt,password='rCd8GEoJZ3WQTk6aNRYop2gpwQ',password2='ue8qsRrJrBLWKr5q3CWn6ake9w',remote=' Users **USER** rclone_scratch cryptfiles '" "-o" "noappledouble" "-o" "modules=iconv,from_code=UTF-8,to_code=UTF-8-MAC"]
fuse: unknown option `password='rCd8GEoJZ3WQTk6aNRYop2gpwQ''
2023/06/16 11:11:49 ERROR : Encrypted drive ':crypt{vOUl7}:': Mount failed
2023/06/16 11:11:49 Fatal error: failed to mount FUSE fs: mount stopped before callin
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
In the forum kapitainsky bisected it down to this commit

3567a47258c7c29aac3870add592975564122f22 is the first bad commit

```
commit 3567a47258c7c29aac3870add592975564122f22
Author: Nick Craig-Wood <nick@craig-wood.com>
Date:   Fri Apr 28 11:58:49 2023 +0100

    fs: make ConfigString properly reverse suffixed file systems
    
    Before this change we renamed file systems with overridden config with
    {suffix}.
    
    However this meant that ConfigString produced a value which wouldn't
    re-create the file system.
    
    This uses an internal hash to keep note of what config goes which
    which {suffix} in order to remake the config properly.

 fs/newfs.go      | 35 ++++++++++++++++++++++++++++++-----
 fs/newfs_test.go | 43 +++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 73 insertions(+), 5 deletions(-)
 create mode 100644 fs/newfs_test.go
```

So what is happening is when we are making the volume name we are getting a whole lot of extra parameters.

Can you try this please? I think it should fix the issue

[v1.63.0-beta.7073.987da19d8.fix-7063-volume-name](https://beta.rclone.org/branch/fix-7063-volume-name/v1.63.0-beta.7073.987da19d8.fix-7063-volume-name/) on branch [fix-7063-volume-name](https://github.com/rclone/rclone/tree/fix-7063-volume-name) (uploaded in 15-30 mins)

**Comment by Jwink3101** [comment]
On macOS 12.6.6 (AMD64) with FUSE-T, this now works. 

The original test was a different OS, processor, and used macFUSE. I verified that both failed previously though the FUSE-T one didn't cause as much "collateral damage".

I still need to test with macOS 13.x (ARM) with macFUSE tonight but I suspect it'll work

**Comment by Jwink3101** [comment]
Tested on the original machine and everything works! Thanks!

**Comment by ncw** [comment]
I decided to change the approach here as I didn't like the passwords in the log.

Can you give this a go instead please?

[v1.63.0-beta.7074.f68475027.fix-7063-volume-name](https://beta.rclone.org/branch/fix-7063-volume-name/v1.63.0-beta.7074.f68475027.fix-7063-volume-name/) on branch [fix-7063-volume-name](https://github.com/rclone/rclone/tree/fix-7063-volume-name) (uploaded in 15-30 mins)

**Comment by kapitainsky** [comment]
> The original test was a different OS, processor, and used macFUSE. I verified that both failed previously though the FUSE-T one didn't cause as much "collateral damage".

Yes this is what I noticed too - unfortunately macFUSE nature working in kernel space is prone to disastrous consequences in case of any issue. And given Apple kexts policies it is "living" legacy anyway.

**Comment by kapitainsky** [comment]
> Can you give this a go instead please?

FYI - `MySecretPassword` is still in the log

```
$ ./rclone mount cryptx: ~/Downloads/mount --allow-non-empty -vv
2023/06/22 19:58:31 DEBUG : rclone: Version "v1.63.0-beta.7074.f68475027.fix-7063-volume-name" starting with parameters ["./rclone" "mount" "cryptx:" "/Users/kptsky/Downloads/mount" "--allow-non-empty" "-vv"]
2023/06/22 19:58:31 DEBUG : Creating backend with remote "cryptx:"
2023/06/22 19:58:31 DEBUG : Using config file from "/Users/kptsky/.config/rclone/rclone.conf"
2023/06/22 19:58:31 DEBUG : Setting type="crypt" for "cryptx" from environment variable RCLONE_CONFIG_CRYPTX_TYPE
2023/06/22 19:58:31 DEBUG : Setting remote="/Users/kptsky/Downloads/data" for "cryptx" from environment variable RCLONE_CONFIG_CRYPTX_REMOTE
2023/06/22 19:58:31 DEBUG : Setting password="MySecretPassword" for "cryptx" from environment variable RCLONE_CONFIG_CRYPTX_PASSWORD
2023/06/22 19:58:31 DEBUG : cryptx: detected overridden config - adding "{bd99F}" suffix to name
2023/06/22 19:58:31 DEBUG : Setting remote="/Users/kptsky/Downloads/data" for "cryptx" from environment variable RCLONE_CONFIG_CRYPTX_REMOTE
2023/06/22 19:58:31 DEBUG : Setting password="MySecretPassword" for "cryptx" from environment variable RCLONE_CONFIG_CRYPTX_PASSWORD
2023/06/22 19:58:32 DEBUG : Creating backend with remote "/Users/kptsky/Downloads/data"
2023/06/22 19:58:32 DEBUG : fs cache: renaming cache item "cryptx:" to be canonical "cryptx{bd99F}:"
2023/06/22 19:58:32 INFO  : Encrypted drive 'cryptx{bd99F}:': poll-interval is not supported by this remote
2023/06/22 19:58:32 DEBUG : Mounting on "/Users/kptsky/Downloads/mount" ("cryptx{bd99F}")
2023/06/22 19:58:32 DEBUG : Adding "-o modules=iconv,from_code=UTF-8,to_code=UTF-8-MAC" for macOS
2023/06/22 19:58:32 DEBUG : Encrypted drive 'cryptx{bd99F}:': Mounting with options: ["-o" "attr_timeout=1" "-o" "fsname=cryptx{bd99F}:" "-o" "subtype=rclone" "-o" "max_readahead=131072" "-o" "atomic_o_trunc" "-o" "daemon_timeout=600" "-o" "volname=cryptx{bd99F}" "-o" "noappledouble" "-o" "modules=iconv,from_code=UTF-8,to_code=UTF-8-MAC"]
2023/06/22 19:58:32 DEBUG : Encrypted drive 'cryptx{bd99F}:': Init: 
```

**Comment by darthShadow** [comment]
This was a issue on linux too using mount2 and the above commit fixed the issue there too.

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.63

@kapitainsky wrote
> FYI - MySecretPassword is still in the log
> 
> 2023/06/22 19:58:31 DEBUG : Setting password="MySecretPassword" for "cryptx" from environment variable RCLONE_CONFIG_CRYPTX_PASSWORD

That isn't a new thing. Rclone DEBUGs the value of every environment variable it reads as part of the config.

Do you think we should change this? If so please open a [new issue](https://github.com/rclone/rclone/issues/new/choose) and I'll make the changes there.

I have a plan to mark some variables as sensitive and some as not so that we can make an `rclone config redacted` command to print the config sensibly redacted for support purposes. So if we had that too then we could only print the values of the ones which weren't sensitive.

Perhaps we should stop it printing the values now, and reconsider when we have a sensitive flag.
